c++/src/kj/compat/gzip.h

```diff
@@ -83,26 +83,35 @@ class GzipAsyncInputStream final: public AsyncInputStream {
 
 class GzipAsyncOutputStream final: public AsyncOutputStream {
 public:
-  GzipAsyncOutputStream(AsyncOutputStream& inner, int compressionLevel = Z_DEFAULT_COMPRESSION);
+  GzipAsyncOutputStream(AsyncOutputStream& inner, kj::Maybe<int> compressionLevel = Z_DEFAULT_COMPRESSION);
   ~GzipAsyncOutputStream() noexcept(false);
   KJ_DISALLOW_COPY(GzipAsyncOutputStream);
+  GzipAsyncOutputStream(GzipAsyncOutputStream&&) = default;
```

Comment:
Oach, that's a good catch. Can't say that I like enum approach too much, but it's indeed safer in this regard.

Reply:
I suppose I could equally just declare a const `DECOMPRESS` to alias `nullptr`?

Reply:
Or even go back from `kj::Maybe<int>` to `int` and define own `Z_DECOMPRESS = -2` or something like that... It's all hacky though.

Reply:
Well, I like type-safety, so I'd be in favor of overloading.

Reply:
Note that in modern C++ you can define one constructor to call another, like:

    GzipAsyncOutputStream(AsyncOutputStream& inner, decltype(DECOMPRESS))
        : GzipAsyncOutputStream(inner, nullptr) {}
    GzipAsyncOutputStream(AsyncOutputStream& inner, int compressionLevel)
        : GzipAsyncOutputStream(inner, Maybe<int>(compressionLevel)) {}

Then make the version that accepts `Maybe<int>` be private.

Reply:
Regarding last comment - sure, I've considered that, but it seems to provide little benefit to hide a version with `Maybs<int>` and provide overload from `int` when the former can just stay exposed, and such conversion from `int` to `Maybe<int>` happens automatically anyway.

Although, if I do add decompress variant via an enum, then `Maybe<int>` version probably doesn't need to exist anyway as these constructors will be independent.

Reply:
Switched to the suggested enum variant and removed move constructors.

Reply:
I'd prefer to hide the `Maybe<int>` version. Having two ways to express the same thing makes the API more confusing. Also, there is weirdness around the fact that the literal value `0` is considered by C++ to be convertable to `nullptr_t`. E.g. if it were `Maybe<uint>` and you tried to pass the literal `0`, the compiler would complain that it is ambiguous. Technically this doesn't happen for `Maybe<int>` because the literal `0` is considered to be of type `int`... but I'd rather avoid these issues.

Reply:
As I said, it's already not needed anymore after the constructor split, so `Maybe` version now exists only in the private helper.

Reply:
Oops, sorry! I misread your message as saying you wanted to keep the `Maybe<int>` constructor and I didn't actually look at the code, which was silly of me.

The code looks great, thanks.